client/src/reducers/teaReducers.js

```diff
@@ -1,20 +1,37 @@
-import { Set, Map } from "immutable";
-
-export default (state = [], action) => {
+export default (state = { allTeas: {}, teaIDs: [] }, action) => {
   switch (action.type) {
     case "ADD_TEA":
-      return [...state, action.payload];
-    case "DELETE_TEA":
-      return state.filter(t => t.id !== action.payload.id);
+      let addAllTeas = {
+        ...state.allTeas,
+        [action.payload.id]: action.payload
+      };
+      let addIDsArr = [...state.teaIDs, action.payload.id];
+
+      return { ...state, allTeas: addAllTeas, teaIDs: addIDsArr };
+
     case "EDIT_TEA":
-      return state.map(t => (t.id === action.payload.id ? action.payload : t));
+      let updateAllTeas = {
+        ...state.allTeas,
+        [action.payload.id]: action.payload
+      };
+
+      return { ...state, allTeas: updateAllTeas };
+
+    case "DELETE_TEA":
+      let deleteIDsArr = state.teaIDs.filter(teaID => teaID !== action.payload);
+
+      return { ...state, teaIDs: deleteIDsArr };
+
     case "GET_TEAS":
-      // TODO: Data must be normalized.
-      // I wasn't planning on using immutable but it solved my problem here really well
-      // TODO figure out this TS error
-      return Set([...state, ...action.payload])
-        .map(Map)
-        .toJS();
+      let getAllTeas = {};
+      let getIDsArr = [];
+
+      action.payload.map(tea => {
+        getAllTeas[tea.id] = tea;
+        getIDsArr.push(tea.id);
+      });
```

Comment:
`map` should only be used when you're returning an array. Use `forEach` here instead.